src/App.tsx

```diff
@@ -3,33 +3,53 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.target.value,
+    });
+  };
 
   render() {
+    const visibleMovies = moviesFromServer.filter(movie => {
+      const title = movie.title.toLowerCase();
+      const description = movie.description.toLowerCase();
+
+      return title.includes(this.state.query.toLowerCase())
+        || description.includes(this.state.query.toLowerCase());
+    });
```

Comment:
```suggestion
    const query = this.state.query.toLocaleLowerCase();

    const visibleMovies = moviesFromServer.filter(({ title, description }) => {
      const lowTitle = title.toLowerCase();
      const lowDescr = description.toLocaleLowerCase();

      return lowTitle.includes(query) || lowDescr.includes(query);
    });
```
Better for readability